packages/monaco/src/browser/monaco-diff-nagivator-factory.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { DiffNavigator } from '@theia/editor/lib/browser';
+
+import IStandaloneDiffEditor = monaco.editor.IStandaloneDiffEditor;
+import IDiffNavigatorOptions = monaco.editor.IDiffNavigatorOptions;
+
+export class MonacoDiffNavigatorFactory {
+
+    static nullNavigator = <DiffNavigator>{
+        canNavigate: () => false,
+        hasNext: () => false,
+        hasPrevious: () => false,
+        next: () => undefined,
+        previous: () => undefined,
+        revealFirst: false,
+        dispose: () => undefined,
+    };
+
+    createdDiffNavigator(editor: IStandaloneDiffEditor, options: IDiffNavigatorOptions): DiffNavigator {
+        const navigator = monaco.editor.createDiffNavigator(editor, options);
+        return <DiffNavigator>{
+            canNavigate: () => navigator.canNavigate(),
+            hasNext: () => navigator.nextIdx + 1 < navigator.ranges.length,
```

Comment:
can `nextIdx` be uninitialized at the beginning? Should not we call some initialization method, like `_initIdx`?

Reply:
done